service/share/full_availability.go

```diff
@@ -13,19 +13,19 @@ import (
 // recovery technique. It is considered "full" because it is required
 // to download enough shares to fully reconstruct the data square.
 type fullAvailability struct {
-	getter format.NodeGetter
+	service format.DAGService
```

Comment:
Just a question: what triggered changing this to point to a DAGService instead of a NodeGetter? From reading the changes it seems that everywhere only the the NodeGetter part is used. Or am I missing sth?

Reply:
Because now, we are committing shares to ipfs after successful repairing. And it requires ipld.NodeAdder.